pkg/render/node.go

```diff
@@ -589,6 +589,10 @@ func (c *nodeComponent) nodeDaemonset(cniCfgMap *v1.ConfigMap) *apps.DaemonSet {
 		initContainers = append(initContainers, c.flexVolumeContainer())
 	}
 
+	if c.bpfDataplane() {
```

Comment:
We only need this with AKS so we should only add this container when using ProviderAKS is set.

Reply:
I think it's more broadly useful than that so I'd prefer to leave it in.  On Ubuntu 20.04 and Bottlerocket, it's a no-op but it covers Ubuntu 18.04, which is still pretty common on-prem (and maybe in other clouds that we haven't tested with operator yet).